yarn-project/merkle-tree/src/interfaces/indexed_tree.ts

```diff
@@ -18,14 +18,29 @@ export interface LeafData {
   nextValue: bigint;
 }
 
+/**
+ * A Merkle tree that supports efficient lookup of leaves by value.
```

Comment:
Isn't it by index? I believe we're doing an exhaustive search today when we search by value.

Reply:
Github copilot misbehaved here 😂 Will fix it. Thanks